src/components/App.js

```diff
@@ -1,17 +1,167 @@
 import React from 'react';
+import Search from './Search.js';
+import SearchResults from './SearchResults.js';
+import MovieDisplay from './MovieDisplay.js';
+import Pagination from './Pagination.js';
 
 class App extends React.Component {
   constructor(){
+
     super();
+
+    this.receiveInput = this.receiveInput.bind(this)
+    this.receiveSubmit = this.receiveSubmit.bind(this)
+    this.receiveFocus = this.receiveFocus.bind(this)
+    this.receiveBlur = this.receiveBlur.bind(this)
+    this.getQueriedMovies = this.getQueriedMovies.bind(this)
+    this.receiveMovieID = this.receiveMovieID.bind(this)
+    this.displayCurrentMovie = this.displayCurrentMovie.bind(this)
+    this.receivePageNumber = this.receivePageNumber.bind(this)
+
+    this.state = {
+      baseURL: 'http://www.omdbapi.com/?apikey=2454706d',
+      searchQuery: '',
+      searchDisplay: false,
+      results: [],
+      error: '',
+      currentMovie: false,
+      currentPage: '',
+      pages: ''
+    }
+  }
+
+  // receive user input from Search component
+  receiveInput (text) {
+    this.setState({
+      searchQuery: text,
+      searchDisplay: text.length > 0
+    }, () => {
+        if (text.length > 2) { 
+          this.getQueriedMovies ()
+        }
+      })
+  }
+
+  // receive form submission from Search component
+  receiveSubmit () {
+    this.getQueriedMovies ()
+  }
+
+  // receive focus event from Search component
+  receiveFocus () {
+    this.setState({ searchDisplay: this.state.searchQuery.length > 0})
```

Comment:
If `searchDisplay` is computed in render method, we can set a `focused` property in state here